cylc/flow/platforms.py

```diff
@@ -67,7 +67,7 @@ def log_platform_event(
 def get_platform(
     task_conf: Union[str, None] = None,
     task_id: str = UNKNOWN_TASK,
-    bad_hosts: Optional[Set[str]] = None
+    bad_hosts: Optional[List[str]] = None
```

Comment:
Do `bad_hosts` need to be ordered too or just `good_hosts`?

Reply:
No, but it makes more sense to me if hosts and badhosts are the same type.